### PR TITLE
83 - ods component - text field

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    remember(leadingIconChecked, selectedDisplayType, selectedTrailingElement) {
+        TextFieldCustomizationState(leadingIconChecked, selectedDisplayType, selectedTrailingElement)
+    }
+
+class TextFieldCustomizationState(
+    val leadingIconChecked: MutableState<Boolean>,
+    val selectedState: MutableState<DisplayType>,
+    val selectedTrailingElement: MutableState<TrailingElement>
+) {
+    companion object {
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt:36]: Do we really need a companion object here ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        enum class DisplayType {
+            DEFAULT, ERROR, DISABLED
+        }
+
+        enum class TrailingElement {
+            NONE, ICON, TEXT
+        }
+    }
+
+    val enabled
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt:46]: Maybe it should be renamed to `isEnabled` ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/textfields/SubComponentTextFieldFilled.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import com.orange.ods.demo.ui.components.utilities.clickOnElement
+
+@ExperimentalMaterialApi
+@Composable
+fun TextFieldFilledContent(customizationState: TextFieldCustomizationState) {
+    var text by rememberSaveable { mutableStateOf("") }
+    val context = LocalContext.current
+    val trailingIconName = stringResource(id = R.string.component_element_trailing)
+
+    OdsTextField(
```
Comment (florentmaitre) [inline demo/src/main/java/com/orange/ods/demo/ui/components/textfields/SubComponentTextFieldFilled.kt:37]: Should we rename it to `OdsFilledTextField` ?

File: demo/src/main/java/com/orange/ods/demo/ui/components/Component.kt [hunk 4]
```diff
@@ -80,7 +86,8 @@ val components = listOf(
     Component.Progress,
     Component.RadioButtons,
     Component.Sliders,
-    Component.Switches
+    Component.Switches,
+    Component.TextFields
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/Component.kt:90]: I saw that but I had this warning `Call uses reflection API which is not found in compilation classpath. Make sure you have kotlin-reflect.jar in the classpath` and I didn't know if it was a good idea to add this. What do you think?

File: demo/src/main/java/com/orange/ods/demo/ui/components/utilities/ComponentCustomizationSheet.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+@Composable
+fun ComponentCustomizationSheetHeader(onClick: () -> Unit = {}) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(56.dp)
+            .clickable(onClick = onClick)
+    ) {
+        OdsTextSubtitle1(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/utilities/ComponentCustomizationSheet.kt:40]: We are waiting the bottom sheets specifications from Stephen. He is working on it. I think we will have to update bottom sheets after that cause there is no specification yet.

File: demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt (hunk 1 above)
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/textfields/TextFieldCustomizationState.kt:36]: no you're right

File: demo/src/main/java/com/orange/ods/demo/ui/components/textfields/SubComponentTextFieldFilled.kt (hunk 3 above)
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/ui/components/textfields/SubComponentTextFieldFilled.kt:37]: I called it `OdsTextField` to match with the framework composables names. The Jetpack Compose composables  are `TextField` and `OutlinedTextField`.

File: lib/src/main/java/com/orange/ods/compose/component/textfield/OdsOutlinedTextField.kt [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+ * @param singleLine when set to true, this text field becomes a single horizontally scrolling
+ * text field instead of wrapping onto multiple lines. The keyboard will be informed to not show
+ * the return key as the [ImeAction]. Note that [maxLines] parameter will be ignored as the
+ * maxLines attribute will be automatically set to 1.
+ * @param maxLines the maximum height in terms of maximum number of visible lines. Should be
+ * equal or greater than 1. Note that this parameter will be ignored and instead maxLines will be
+ * set to 1 if [singleLine] is set to true.
+ */
+@Composable
+fun OdsOutlinedTextField(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/textfield/OdsOutlinedTextField.kt:91]: We are using two different composables of Jetpack compose (`TextField` and `OutlinedTextField`) so I think the composables offered by the ODS lib should be the same. But I agree that there is duplicated code. I'm going to factorize it.

**Pull request merged**